packages/framework/src/Component/Elasticsearch/IndexDefinition.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Component\Elasticsearch;
+
+use Shopsys\FrameworkBundle\Component\Elasticsearch\Exception\ElasticsearchIndexException;
+
+class IndexDefinition
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractIndex
+     */
+    protected $index;
+
+    /**
+     * @var string
+     */
+    protected $definitionsDirectory;
+
+    /**
+     * @var string
+     */
+    protected $indexPrefix;
+
+    /**
+     * @var int
+     */
+    protected $domainId;
```

Comment:
in my opinion Index cannot be depended on domain. ... I'm just saying...

Reply:
Actually it is. Each index is created per domain by design. That is because you want to have different configuration according to the language. I know domain != language, but it is the way we work with it and we are not changing it in this PR. If you have any improvement tips please let us know we want to hear it.